lib/jobs/sftp-job.js

```diff
@@ -0,0 +1,119 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+module.exports = sftpJobFactory;
+
+var Client = require('ssh2').Client;
+di.annotate(sftpJobFactory, new di.Provide('Job.Sftp'));
+di.annotate(sftpJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Promise',
+    'Services.Encryption',
+    'Services.Waterline'
+));
+
+function sftpJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Promise,
+    cryptService,
+    waterline
+) {
+    var logger = Logger.initialize(sftpJobFactory);
+
+    function SftpJob(options, context, taskId) {
+        SftpJob.super_.call(this, logger, options, context, taskId);
+        assert.string(options.fileSource);
+        assert.string(options.fileDestination);
+
+        assert.string(this.context.target);
+        this.nodeId = this.context.target;
+        this.timeout = options.timeout;
+        this.fileSource = options.fileSource;
+        this.fileDestination = options.fileDestination;
+    }
+    util.inherits(SftpJob, BaseJob);
+
+    SftpJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.needByIdentifier(this.context.target)
```

Comment:
Use `self.nodeId` instead of `this.context.target`? If not, then the assignment of `self.nodeId` (https://github.com/RackHD/on-tasks/pull/163/files#diff-9c80485bef6f959152431eaa9b8160f9R37) will have no reference.